### PR TITLE
Update simulation_parsing.py

### LINKED ISSUE #688: Divide by zero error happening in universal_cartesian for long-period comets
I'm getting the following error for one of the LPC DP0.3 input files when running on Kelvin:

```
Traceback (most recent call last):
  File "/users/3057027/.conda/envs/sorcha/bin/sorcha", line 8, in <module>
    sys.exit(main())
  File "/users/3057027/sorcha/src/sorcha/sorcha.py", line 498, in main
    runLSSTSimulation(args, configs, pplogger)
  File "/users/3057027/sorcha/src/sorcha/sorcha.py", line 184, in runLSSTSimulation
    observations = create_ephemeris(orbits_df, filterpointing, args, configs)
  File "/users/3057027/sorcha/src/sorcha/ephemeris/simulation_driver.py", line 40, in create_ephemeris
    sim_dict = generate_simulations(ephem, gm_sun, orbits_df)
  File "/users/3057027/sorcha/src/sorcha/ephemeris/simulation_setup.py", line 94, in generate_simulations
    x, y, z, vx, vy, vz = sp.parse_orbit_row(row, epoch, ephem, sun_dict, gm_sun)
  File "/users/3057027/sorcha/src/sorcha/ephemeris/simulation_parsing.py", line 29, in parse_orbit_row
    ecx, ecy, ecz, dx, dy, dz = universal_cartesian(
ZeroDivisionError: division by zero
```

The function `universal_cartesian` is in `orbit_conversion_utilities.py`.

This seems to be machine-dependent (@bernardinelli tried this on his own machine with no problems). 

I'm going to try to narrow down exactly which input line is triggering this. I'm also going to try commenting out the numba decorators on Pedro's recommendation to see if I can get a more informative error message. May also see if I can get the error on my laptop.

The input files in question were emailed around the mailing list a while back. Let me know if you can't find them and want them :)

### LINKED ISSUE #695: change t_p to t_p_MJD_TDB
we should be very clear that t_p is specified in MJD_TBD like epoch

### PR DESCRIPTION
**Pull request by bernardinelli**
Adds MJD to JD conversion in `parse_orbit_row`

Fixes #688 
Fixes #695 

Describe your changes.

## Review Checklist for Source Code Changes

- [x ] Does pip install still work?
- [ ] Have you written a unit test for any new functions?
- [x ] Do all the units tests run successfully?
- [x ] Does Sorcha run successfully on a test set of input files/databases?
- [x ] Have you used black on the files you have updated to confirm python programming style guide enforcement?

### CODE REVIEW

File: src/sorcha/ephemeris/simulation_parsing.py [hunk 1]
```diff
@@ -21,31 +21,40 @@ def mjd_tai_to_epoch(mjd_tai):
...
                 row["argPeri"] * np.pi / 180.0,
-                row["t_p"],
-                epochMJD_TDB,
+                t_p,
+                epochJD_TDB,
             )
         elif orbit_format == "BCOM":
+            # JD to MJD conversion if necessary
+            t_p = row["t_p"]
+            if t_p < 2400000.5:
```
Comment (mschwamb) [inline src/sorcha/ephemeris/simulation_parsing.py:47]: Can we do this when we do the epoch conversion with the [one for epoch ](https://github.com/dirac-institute/sorcha/blob/846b251eb4412800dcb2c00bf2b63362e296997b/src/sorcha/ephemeris/simulation_setup.py#L93C8-L95C1)and change t_p here/ in the structure so it is tp_JD_TDB so its crystal clear what the units are.  
Good catch on the time labels being incorrect here.

**Review by mschwamb — CHANGES_REQUESTED** [review]

File: src/sorcha/ephemeris/simulation_parsing.py (hunk 1 above)
Comment (mschwamb) [reply src/sorcha/ephemeris/simulation_parsing.py:47]: Actually on second thought it's fine to do it the conversion here, but let's be super clear about labeling variables.  t_p variable should have that is t_p_JD_TDB so it's crystal clear what unit is is in. Note there's ticket/issue in to rename t_p for the orbit inputs so it's explicitly t_p_MD_TDB 

We're requiring that all orbits be input in MJD for epoch and T0 do we need to do this test?  of the t_p < 2400000.5 and instead just to the conversion. If we're really worried that this the code could  give a warning in the log file if tp > 2400000.5: add a message to the log file saying the user should check their inputs to ensure the inputs are in MJD.


Question for @matthewholman  do we need to do the test the epoch and adjust in https://github.com/dirac-institute/sorcha/blob/main/src/sorcha/ephemeris/simulation_setup.py#L92-#L94  when wer're explicitly sending in MJD? I think in

Comment (bernardinelli) [reply src/sorcha/ephemeris/simulation_parsing.py:47]: yeah - we can just assume that the user will supply a mjd and do the conversion anyway. I like the idea of adding something to the log file if it looks like something is in JD.

**Comment by bernardinelli** [comment]
Also fixing #695 in this PR

**Comment by mschwamb** [comment]
@bernardinelli  can you install black and black the files. It's the required linter that makes things look pretty. It's the first test that's failing

**Comment by bernardinelli** [comment]
> @bernardinelli can you install black and black the updated files? It's the required linter that makes things look pretty. It's the first test that's failing

I did run black locally prior to pushing - no idea why it's failing that test!

**Comment by mschwamb** [comment]
I'd try running it again on simulation_parsing.py

**Comment by bernardinelli** [comment]
wow - re-running black removed a space at the end of a line...!

**Review by mschwamb — APPROVED** [review]

**Pull request merged**